Scripts/RemediationScripts/Remediate-AnonymousAccessOnContainers.ps1

```diff
@@ -316,10 +326,20 @@ function Remove-AnonymousAccessOnContainers
                 if ($totalStgWithEnableAllowBlobPublicAccess -gt 0)
                 {
                     # Creating the log file
-                    
-                    Write-Host "Backing up config of storage account(s) details. Please do not delete this file without this file you won't be able to rollback any changes done through remediation script." -ForegroundColor $([Constants]::MessageType.Info)
-                    $stgWithEnableAllowBlobPublicAccess | ConvertTo-json | out-file "$($folderpath)\DisabledAllowBlobPublicAccess.json"  
-                    Write-Host "Path: $($folderpath)\DisabledAllowBlobPublicAccess.json"     
+                    if ($DryRun)
+                    {
+                        Write-Host "Exporting configurations of Storage account(s) having 'Allow Blob Public Access' enabled. You may want to use this CSV as a pre-check before actual remediation." -ForegroundColor Cyan
+                        $stgWithEnableAllowBlobPublicAccess | Export-CSV -Path "$($folderpath)\StorageWithEnabledAllowBlobPublicAccess.csv" -NoTypeInformation
+                        Write-Host "Path: $($folderPath)DisabledAllowBlobPublicAccess.csv"
+                        return;
+                    }
+                    else
+                    {
+                        Write-Host "Backing up config of storage account(s) details. Please do not delete this file. Without this file you won't be able to rollback any changes done through remediation script." -ForegroundColor $([Constants]::MessageType.Info)
+                        $stgWithEnableAllowBlobPublicAccess | ConvertTo-json | out-file "$($folderpath)\DisabledAllowBlobPublicAccess.json"  
+                        Write-Host "Path: $($folderpath)\DisabledAllowBlobPublicAccess.json"
+                    }
+                         
```

Comment:
ResourceResolver check if this is loaded or not?

Reply:
At line # 306 Use Get-Member instead.

Reply:
Added try-catch for ResourceResolver, if module not loaded.

Reply:
Updated condition with Get-Member.